src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectly.cs

```diff
@@ -85,7 +85,16 @@ private static void OnCompilationStart(CompilationStartAnalysisContext context)
                 }
 
                 var properties = new Dictionary<string, string?> { [MethodPropertyKey] = invocation.TargetMethod.Name }.ToImmutableDictionary();
-                operationContext.ReportDiagnostic(invocation.CreateDiagnostic(Rule, properties));
+                IOperation reportDiagnosticOn = invocation;
+
+                // If the parent of invocation is IConditionalAccessOperation then we need to report
+                // diagnostic on it in order to be consistent with diagnostic location.
+                // If we don't do that the diagnostic is reported here: x?[|.Last()|],
+                // while the expected placement is [|x?.Last()|]
+                if (invocation is { Parent: IConditionalAccessOperation conditionalAccess })
+                    reportDiagnosticOn = conditionalAccess;
+
+                operationContext.ReportDiagnostic(reportDiagnosticOn.CreateDiagnostic(Rule, properties));
```

Comment:
I feel that if we leave the diagnostic location as is, it will be easier to implement the codefix. But can't tell for sure without trying it out.

Reply:
There is a `TryExtractInvocationOperation` method in the codefix that retrieves invocation operation from the diagnostic considering that trait (19 lines of code, 2 of which are empty). It isn't a big deal.